osu.Game/Screens/OnlinePlay/Components/SelectionPollingComponent.cs

```diff
@@ -30,19 +30,22 @@ protected override Task Poll()
 
             var tcs = new TaskCompletionSource<bool>();
 
-            pollReq?.Cancel();
-            pollReq = new GetRoomRequest(room.RoomID.Value.Value);
+            lastPollRequest?.Cancel();
 
-            pollReq.Success += result =>
+            var req = new GetRoomRequest(room.RoomID.Value.Value);
+
+            req.Success += result =>
             {
                 result.RemoveExpiredPlaylistItems();
                 RoomManager.AddOrUpdateRoom(result);
                 tcs.SetResult(true);
             };
 
-            pollReq.Failure += _ => tcs.SetResult(false);
+            req.Failure += _ => tcs.SetResult(false);
+
+            API.Queue(req);
 
-            API.Queue(pollReq);
+            lastPollRequest = req;
```

Comment:
Is there a reason for these changes? Or are they just something that you've done along the way because why not? As far as I can tell they should not have any difference, right?

Reply:
Does the commit message explain enough?

![20220128 161028 (Safari)](https://user-images.githubusercontent.com/191335/151503149-8ab624a5-25a6-4977-9456-601a55673f8b.png)

Was just double-safety.

Reply:
Can you explain how it's more safe? I just want to know what I'm missing for future implementations, because as it stands I would do it exactly the same way it was done previously.

In-fact isn't this technically more unsafe? If we're talking about the % chance that `lastPollRequest` is cancelled if this method is called asynchronously, then this has a lower % chance to cancel the request and a higher % chance to fire duplicates.

Again I'm just trying to know the thought process.

Reply:
Because it was referencing a field, this caused it to be thread unsafe. Consider two threads running the method:

- Thread1 constructs a request
- Thread2 constructs a request (and overwrites the first)
- Thread1 queues the request
- Thread2 queues the request <- point of failure

Reply:
Right, I see. So the problem is enqueueing the same request object in both paths. Okay.